src/wasm2js.h

```diff
@@ -341,6 +341,9 @@ Ref Wasm2JSBuilder::processWasm(Module* wasm, Name funcName) {
     // TODO: only legalize if necessary - emscripten would already do so, and
     //       likely other toolchains. but spec test suite needs that.
     runner.add("legalize-js-interface");
+    // Before lowering non-JS operations we can rewrite some instructions which
+    // may simplify next passes
+    runner.add("pre-js-rewriter");
```

Comment:
I would name the pass "pre-js-rewrite" to make it more of a verb instead of a noun.